src/probability/extrealScript.sml

```diff
@@ -9463,6 +9463,268 @@ Proof
  >> rw [reduced_metric_lt_1]
 QED
 
+(************************************************************************)
+(*   Miscellaneous Results (generally for use in descendent theories)   *)
+(************************************************************************)
+
+(*  I add these results at the end
+      in order to manipulate the simplifier without breaking anything
+      - Jared Yeager                                                    *)
+
+(*** Simplification Definitions ***)
+
+val name_to_thname = fn s => ({Thy = "extreal", Name = s}, DB.fetch "extreal" s);
+
+val mk_local_simp = augment_srw_ss o single o
+  simpLib.rewrites_with_names o single o name_to_thname;
+
+(*** Basic Theorems ***)
+
+Theorem normal_0:
+    Normal 0 = 0
+Proof
+    simp[extreal_eq_zero]
+QED
+
+Theorem normal_1:
+    Normal 1 = 1
+Proof
+    simp[extreal_of_num_def]
+QED
+
+Theorem normal_minus1:
+    Normal (-1) = -1
+Proof
+    ‘Normal (-1) = -(Normal 1)’ suffices_by simp[normal_1] >> simp[extreal_ainv_def]
+QED
+
+Theorem extreal_le_simp:
```

Comment:
This should be a permanent `[simp]`.

Reply:
I agree, about this and the following two. That said, I suspect they will break a number of existing proofs, so I'll tackle them last.
I'm guessing I can use something like diminish to just remove them from consideration for areas where they would cause problems, and augment them back in afterwards.

Reply:
I'd hope that automatic rules for `<` and `<=` would just make everyone's life simpler. My annotations may not have been clear enough, but I certainly wasn't suggesting that the rewrites for numerals inside and outside `Normal` should be `simp`.

Reply:
I absolutely agree. I just suspect that there are some proofs that produced subgoals that having these simps will now automatically solve. Or some assumption gets simplified and so a dependent drule or irule doesn't work right. I suspect something will break. So I wanted to get easier stuff addressed first.

That said, perhaps I will tackle these while I wait for judgement to be rendered on the various `(EXT)REAL_{SUM,PROD}_IMAGE_{EMPTY,SING,COUNT_{ZERO,ONE}}` theorems.

(Also, thankfully, I didn't misunderstand about the `Normal 0 = 0` type theorems. Otherwise, I would have commented about how many times I've used those backwards.)